ntp-daemon/src/main.rs

```diff
@@ -24,7 +24,7 @@ async fn setup_connection() -> std::io::Result<UdpSocket> {
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error>> {
-    let socket = setup_connection().await?;
+    let socket = ntp_udp::UdpSocket::from_tokio(setup_connection().await?)?;
```

Comment:
Why not add a basic constructor to `ntp_udp::UdpSocket` that takes the source and destination addresses, and move that functionality out of this crate's `setup_connection()`? That seems a simpler design. (I would probably skip having the `from_std()` and `from_tokio()` constructors unless they add value for testing.)